internal/encryption/registry.go

```diff
@@ -0,0 +1,30 @@
+package encryption
+
+// Registry is a holder of KeyProvider and Method implementations. Key providers and methods can register
+// themselves with this registry.
+type Registry struct {
+	KeyProviders map[string]KeyProviderSource
+	Methods      map[string]MethodSource
+}
+
+func NewRegistry() Registry {
+	return Registry{
+		KeyProviders: map[string]KeyProviderSource{
+			"passphrase": func() KeyProvider { return &PassphraseKeyProvider{} },
+		},
+		Methods: map[string]MethodSource{
+			"aes_cipher": func() Method { return &AESCipherMethod{} },
+		},
+	}
+}
+
+type KeyProvider interface {
+	KeyData() ([]byte, error)
+}
+type KeyProviderSource func() KeyProvider
+
+type Method interface {
+	Encrypt([]byte) ([]byte, error)
```

Comment:
This forces every method to correctly implement setting the "encryption" JSON key in its output.

While certainly possible, it might be worth considering keeping this functionality as part of the Flow, compare the Method interface in https://github.com/opentofu/opentofu/pull/1132/files#diff-151f8fee8a175def6beee37ee974a0794b4e39824e906fa1f74c91c8e2a40190